### DISCUSSION TITLE
Environments + Github actions?

### DISCUSSION THREAD

**Discussion by andrewbihl** [community/community#43799] [Actions]
### Select Topic Area

Bug

### Body

Is the "environments" features some sort of avant-garde joke? I am pretty sure I have tried every permutation of YAML configuration attempting to use Github actions with Environment Secrets. The following do not work:
- set the environment of a workflow
- set the environment for running some reusable workflow
- accessing secrets within a reusable workflow
- accessing environment secrets to set as environment variables to a reusable workflow
- passing environment secrets to pass as inputs to a reusable workflow
- print an environment secret

Is there a purpose for this "feature"? Was it turned on by accident? As far as I can tell it only serves the purpose of destroying one day of my life, but I am not sure if I see any point to it beyond that. 

Now I am one day closer to death, and I spent my day fiddling with YAML while reading incorrect documentation. 

Thanks in advance for closing my issue without any response!

**Comment by wechuli** [comment]
Could you please check that you are inheriting secrets from the caller workflow as detailed here:

https://docs.github.com/en/actions/using-workflows/reusing-workflows#passing-inputs-and-secrets-to-a-reusable-workflow

```yml
jobs:
  call-workflow-passing-data:
    uses: octo-org/example-repo/.github/workflows/reusable-workflow.yml@main
    secrets: inherit
```
If you are already doing this, do you mind creating a public repository with a minimal reproduction for me to take a look?

**Reply by andrewbihl** [reply]
I did use that option, it did not work. To be totally honest I do not expect I'll find the time to make a new project and re-create a scenario to show the problem to you.

Do you have a public repository with a minimal reproduction of environments working as expected? Perhaps that would be more helpful; this would provide a way for me to diagnose what is different, serve as a useful template for others to use, and offer some evidence that this feature set has ever worked.

To onlookers just trying to get your work done: I'd still recommend steering clear of anything with environments. It's a rabbit hole with possibly no end.

**Reply by wechuli** [reply]
Here you go - https://github.com/wechuli/environmentsecretwithreusable/tree/main/.github/workflows

The environment is called `dev` and has a secret called PASSWORD. GitHub will mask the secret from the logs as can be seen from https://github.com/wechuli/environmentsecretwithreusable/actions/runs/4009009667/jobs/6883862229
